### PR TITLE
622 AppD lang and translations

### LINKED ISSUE #622: Add language metadata to AppD for accessibility
### Use Case:
* An institution has an app that supports multiple languages. They want to include multiple manifest entries, where they have a different version of the app for each language the app supports.
* Consider an implementation of a desktop agent that contains multiple applications across multiple languages. If an end-user is blind and using a screen reader, the screen reader will be unable to independently determine which applications have which languages. The desktop agent will need to provide that language information, which it must determine for a given application.

For example: 
> A bank uses a desktop agent to provide a financial application. The application has an English and Chinese version. The bank wants their English-language employees to be able to use the English version of all of the applications, and their Chinese-language employees to be able to use the Chinese version of the applications. The bank could provide those language identifiers in their manifest, and then the desktop agent could surface only certain languages that end-users require.

> A bank uses a desktop agent to provide financial applications to users in Canada. They provide copies of their applications in French and in English, and may even have language-specific applications. Let's say one of the users at this bank is blind and uses a screen reader. They are a native English speaker, so their operating system, desktop application, and screen reader are all running in English. Whenever the screen reader encounters the title or contents of the _french_ applications, it pronounces the french words as if they are English, resulting in cognitively-taxing confusion. Further more, let's say this user is also fluent in french. With no metadata provided for the french versus english applications, when the user encounters any of the french apps, the screen reader will pronounce french words using english pronunciation rules. In other words, they become gibberish.

### Workflow Description
If the AppD provided metadata for the language of the app (assuming the title of the app and the contents of the app are all in the same language), the desktop agent could surface that language appropriately.

### Additional Information
HTML contains the global attribute, `lang`. This attribute can be used on any HTML element, and would likely be leveraged by any HTML-based desktop agent. The lang attribute typically uses BCP 47, which includes a 2-digit language code (like "en" or "fr") and optionally a 2-digit country code (like "en-US" or "en-GB").
[HTML lang attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/lang)

The [W3C's Web Application manifest also supports a `lang` attribute](https://w3.org/TR/appmanifest/#lang-member), with [alternative manifest URLs or geolocation and targeting being used to deliver the localized version of the manifest](https://www.w3.org/TR/appmanifest/#internationalization).

### PR DESCRIPTION
**Pull request by kriswest**
resolves #622 
Adds `lang` and `localizedVersions` fields to the appD application record schema to support accessiblity (screenreaders need to know the primary language of the record content) and localisation (of both appD record's content and the applications themselves through alternative launch url/config/hostManifests)

Change agreed at meeting:
- #657

### CODE REVIEW

**Comment by kriswest** [comment]
@greyseer256  If you're in a reviewing mood, could you check https://github.com/finos/FDC3/pull/670 for me - its another one where the appD schema structure has to change (doesn't change the record structure much, but the YAML had to change) so I need to base other PRs off of it (or deal with a lot of conflicts)

File: src/app-directory/specification/appd.yaml [hunk 1]
```diff
@@ -455,6 +444,10 @@ components:
         tooltip:
           type: string
           description: Optional tooltip description e.g. for a launcher
+        lang:
+          type: string
+          pattern: '^[a-z]{2}(-[a-zA-Z0-9]{2,8}){0,1}$'
+          description:  a language tag that specifies the primary language of both application and it's AppD entry, as defined by IETF RFC 5646.
```
Comment (greyseer256) [inline src/app-directory/specification/appd.yaml:450]:
```suggestion
          description:  a language tag that specifies the primary language of both application and its AppD entry, as defined by IETF RFC 5646.
```

File: website/static/schemas/next/app-directory.yaml [hunk 2]
```diff
@@ -455,6 +444,10 @@ components:
         tooltip:
           type: string
           description: Optional tooltip description e.g. for a launcher
+        lang:
+          type: string
+          pattern: '^[a-z]{2}(-[a-zA-Z0-9]{2,8}){0,1}$'
+          description:  a language tag that specifies the primary language of both application and it's AppD entry, as defined by IETF RFC 5646.
```
Comment (greyseer256) [inline website/static/schemas/next/app-directory.yaml:450]:
```suggestion
          description:  a language tag that specifies the primary language of both application and its AppD entry, as defined by IETF RFC 5646.
```

**Review by greyseer256 — APPROVED** [review]
Minor comments, looks good

**Pull request merged**